ScoutSuite/providers/azure/resources/network/network_security_groups.py

```diff
@@ -0,0 +1,129 @@
+            security_rule_dict['destination_address_prefixes'] = destination_address_prefixes
+
+            destination_port_ranges = self._merge_prefixes_or_ports(sr.destination_port_range,
+                                                                    sr.destination_port_ranges)
+            security_rule_dict['destination_port_ranges'] = destination_port_ranges
+            security_rule_dict['destination_ports'] = self._parse_ports(destination_port_ranges)
+
+            security_rule_dict['etag'] = sr.etag
+
+            security_rules[security_rule_dict['id']] = security_rule_dict
+
+        return security_rules
+
+    def _parse_ports(self, port_ranges):
+        ports = set()
+        for pr in port_ranges:
+            if pr == "*":
+                for p in range(0, 65535 + 1):
+                    ports.add(p)
+                break
+            elif "-" in pr:
+                lower, upper = pr.split("-")
+                for p in range(int(lower), int(upper) + 1):
+                    ports.add(p)
+            else:
+                ports.add(int(pr))
+        ports = list(ports)
+        ports.sort()
+        return ports
+
+    def _parse_exposed_ports(self, network_security_group):
+        exposed_ports = set()
+
+        # Sort by priority.
+        rules = network_security_group.default_security_rules + network_security_group.security_rules
+        rules.sort(key=lambda x: x.priority, reverse=True)
+
+        for sr in rules:
+            if sr.direction == "Inbound" and (sr.source_address_prefix == "*"
+                                              or sr.source_address_prefix == "Internet"):
+                port_ranges = self._merge_prefixes_or_ports(sr.destination_port_range,
+                                                                     sr.destination_port_ranges)
+                ports = self._parse_ports(port_ranges)
+                if sr.access == "Allow":
+                    for p in ports:
+                        exposed_ports.add(p)
+                else:
+                    for p in ports:
+                        exposed_ports.discard(p)
+        exposed_ports = list(exposed_ports)
+        exposed_ports.sort()
+        return exposed_ports
+
+    def _merge_prefixes_or_ports(self, port_range, port_ranges):
+        port_ranges = port_ranges if port_ranges else []
+        if port_range:
+            port_ranges.append(port_range)
+        return port_ranges
+
+    def _format_ports(self, ports):
+        port_ranges = []
+        start = None
+        for i in range(0, 65535 + 1):
+            if i in ports:
+                if not start:
+                    start = i
+            else:
+                if start:
+                    if i - 1 == start:
+                        port_ranges.append(str(start))
+                    else:
+                        port_ranges.append(str(start) + "-" + str(i - 1))
+                    start = None
+        return port_ranges
```

Comment:
I feel like it would be worth writing one or two unit tests for this method. It should be pretty quick!